[BUG] test_delta_name_column_mapping_no_field_ids fails on Spark

test_delta_name_column_mapping_no_field_ids failed on Spark 3.2.4 during a recent nightly integration test run.
<details>
<summary>Details</summary>

```
[2024-07-18T18:46:34.572Z] =================================== FAILURES ===================================
[2024-07-18T18:46:34.572Z] [31m[1m_________________ test_delta_name_column_mapping_no_field_ids __________________[0m
[2024-07-18T18:46:34.572Z] [gw2] linux -- Python 3.9.19 /opt/conda/bin/python
[2024-07-18T18:46:34.572Z] 
[2024-07-18T18:46:34.572Z] spark_tmp_path = '/tmp/pyspark_tests//it-test-324-1041-199qh-3mvps-gw2-1303892-382636422/'
[2024-07-18T18:46:34.572Z] 
[2024-07-18T18:46:34.572Z]     [37m@allow_non_gpu[39;49;00m(*delta_meta_allow)[90m[39;49;00m
[2024-07-18T18:46:34.572Z]     [37m@delta_lake[39;49;00m[90m[39;49;00m
[2024-07-18T18:46:34.572Z]     [37m@ignore_order[39;49;00m(local=[94mTrue[39;49;00m)[90m[39;49;00m
[2024-07-18T18:46:34.572Z]     [94mdef[39;49;00m [92mtest_delta_name_column_mapping_no_field_ids[39;49;00m(spark_tmp_path):[90m[39;49;00m
[2024-07-18T18:46:34.572Z]         data_path = spark_tmp_path + [33m"[39;49;00m[33m/DELTA_DATA[39;49;00m[33m"[39;49;00m[90m[39;49;00m
[2024-07-18T18:46:34.572Z]         [94mdef[39;49;00m [92msetup_parquet_table[39;49;00m(spark):[90m[39;49;00m
[2024-07-18T18:46:34.572Z]             spark.range([94m10[39;49;00m).coalesce([94m1[39;49;00m).write.parquet(data_path)[90m[39;49;00m
[2024-07-18T18:46:34.572Z]         [94mdef[39;49;00m [92mconvert_and_setup_name_mapping[39;49;00m(spark):[90m[39;49;00m
[2024-07-18T18:46:34.572Z]             spark.sql([33mf[39;49;00m[33m"[39;49;00m[33mCONVERT TO DELTA parquet.`[39;49;00m[33m{[39;49;00mdata_path[33m}[39;49;00m[33m`[39;49;00m[33m"[39;49;00m)[90m[39;49;00m
[2024-07-18T18:46:34.572Z]             spark.sql([33mf[39;49;00m[33m"[39;49;00m[33mALTER TABLE delta.`[39;49;00m[33m{[39;49;00mdata_path[33m}[39;49;00m[33m` SET TBLPROPERTIES [39;49;00m[33m"[39;49;00m +[90m[39;49;00m
[2024-07-18T18:46:34.572Z]                 [33m"[39;49;00m[33m([39;49;00m[33m'[39;49;00m[33mdelta.minReaderVersion[39;49;00m[33m'[39;49;00m[33m = [39;49;00m[33m'[39;49;00m[33m2[39;49;00m[33m'[39;49;00m[33m, [39;49;00m[33m"[39;49;00m +[90m[39;49;00m
[2024-07-18T18:46:34.572Z]                 [33m"[39;49;00m[33m'[39;49;00m[33mdelta.minWriterVersion[39;49;00m[33m'[39;49;00m[33m = [39;49;00m[33m'[39;49;00m[33m5[39;49;00m[33m'[39;49;00m[33m, [39;49;00m[33m"[39;49;00m +[90m[39;49;00m
[2024-07-18T18:46:34.572Z]                 [33m"[39;49;00m[33m'[39;49;00m[33mdelta.columnMapping.mode[39;49;00m[33m'[39;49;00m[33m = [39;49;00m[33m'[39;49;00m[33mname[39;49;00m[33m'[39;49;00m[33m)[39;49;00m[33m"[39;49;00m)[90m[39;49;00m
[2024-07-18T18:46:34.572Z]         with_cpu_session(setup_parquet_table, {[33m"[39;49;00m[33mspark.sql.parquet.fieldId.write.enabled[39;49;00m[33m"[39;49;00m: [33m"[39;49;00m[33mfalse[39;49;00m[33m"[39;49;00m})[90m[39;49;00m
[2024-07-18T18:46:34.572Z] >       with_cpu_session(convert_and_setup_name_mapping)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] 
[2024-07-18T18:46:34.572Z] [1m[31m../../src/main/python/delta_lake_test.py[0m:138: 
[2024-07-18T18:46:34.572Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2024-07-18T18:46:34.572Z] [1m[31m../../src/main/python/spark_session.py[0m:149: in with_cpu_session
[2024-07-18T18:46:34.572Z]     [94mreturn[39;49;00m with_spark_session(func, conf=copy)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m/opt/conda/lib/python3.9/contextlib.py[0m:79: in inner
[2024-07-18T18:46:34.572Z]     [94mreturn[39;49;00m func(*args, **kwds)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m../../src/main/python/spark_session.py[0m:133: in with_spark_session
[2024-07-18T18:46:34.572Z]     ret = func(_spark)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m../../src/main/python/delta_lake_test.py[0m:132: in convert_and_setup_name_mapping
[2024-07-18T18:46:34.572Z]     spark.sql([33mf[39;49;00m[33m"[39;49;00m[33mCONVERT TO DELTA parquet.`[39;49;00m[33m{[39;49;00mdata_path[33m}[39;49;00m[33m`[39;49;00m[33m"[39;49;00m)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m../../../spark-3.2.4-bin-hadoop3.2/python/pyspark/sql/session.py[0m:723: in sql
[2024-07-18T18:46:34.572Z]     [94mreturn[39;49;00m DataFrame([96mself[39;49;00m._jsparkSession.sql(sqlQuery), [96mself[39;49;00m._wrapped)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-github-1041-3.2.4/jars/spark-3.2.4-bin-hadoop3.2/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py[0m:1321: in __call__
[2024-07-18T18:46:34.572Z]     return_value = get_return_value([90m[39;49;00m
[2024-07-18T18:46:34.572Z] [1m[31m../../../spark-3.2.4-bin-hadoop3.2/python/pyspark/sql/utils.py[0m:111: in deco
[2024-07-18T18:46:34.572Z]     [94mreturn[39;49;00m f(*a, **kw)[90m[39;49;00m
[2024-07-18T18:46:34.572Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2024-07-18T18:46:34.572Z] 
[2024-07-18T18:46:34.572Z] answer = 'xro15425'
[2024-07-18T18:46:34.572Z] gateway_client = <py4j.clientserver.JavaClient object at 0x7fce79955f40>
[2024-07-18T18:46:34.572Z] target_id = 'o83', name = 'sql'
[2024-07-18T18:46:34.572Z] 
[2024-07-18T18:46:34.572Z]     [94mdef[39;49;00m [92mget_return_value[39;49;00m(answer, gateway_client, target_id=[94mNone[39;49;00m, name=[94mNone[39;49;00m):[90m[39;49;00m
[2024-07-18T18:46:34.572Z]     [90m    [39;49;00m[33m"""Converts an answer received from the Java gateway into a Python object.[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    For example, string representation of integers are converted to Python[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    integer, string representation of objects are converted to JavaObject[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    instances, etc.[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    :param answer: the string returned by the Java gateway[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    :param gateway_client: the gateway client used to communicate with the Java[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m        Gateway. Only necessary if the answer is a reference (e.g., object,[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m        list, map)[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    :param target_id: the name of the object from which the answer comes from[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m        (e.g., *object1* in `object1.hello()`). Optional.[39;49;00m
[2024-07-18T18:46:34.572Z]     [33m    :param name: the name of the member from which the answer comes from[39;49;00m
[2024-07-18T18:46:34.573Z]     [33m        (e.g., *hello* in `object1.hello()`). Optional.[39;49;00m
[2024-07-18T18:46:34.573Z]     [33m    """[39;49;00m[90m[39;49;00m
[2024-07-18T18:46:34.573Z]         [94mif[39;49;00m is_error(answer)[[94m0[39;49;00m]:[90m[39;49;00m
[2024-07-18T18:46:34.573Z]             [94mif[39;49;00m [96mlen[39;49;00m(answer) > [94m1[39;49;00m:[90m[39;49;00m
[2024-07-18T18:46:34.573Z]                 [96mtype[39;49;00m = answer[[94m1[39;49;00m][90m[39;49;00m
[2024-07-18T18:46:34.573Z]                 value = OUTPUT_CONVERTER[[96mtype[39;49;00m](answer[[94m2[39;49;00m:], gateway_client)[90m[39;49;00m
[2024-07-18T18:46:34.573Z]                 [94mif[39;49;00m answer[[94m1[39;49;00m] == REFERENCE_TYPE:[90m[39;49;00m
[2024-07-18T18:46:34.573Z] >                   [94mraise[39;49;00m Py4JJavaError([90m[39;49;00m
[2024-07-18T18:46:34.573Z]                         [33m"[39;49;00m[33mAn error occurred while calling [39;49;00m[33m{0}[39;49;00m[33m{1}[39;49;00m[33m{2}[39;49;00m[33m.[39;49;00m[33m\n[39;49;00m[33m"[39;49;00m.[90m[39;49;00m
[2024-07-18T18:46:34.573Z]                         [96mformat[39;49;00m(target_id, [33m"[39;49;00m[33m.[39;49;00m[33m"[39;49;00m, name), value)[90m[39;49;00m
[2024-07-18T18:46:34.573Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o83.sql.[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1254.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1254.0 (TID 15424) (100.111.75.6 executor 0): java.lang.NoSuchMethodError: org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ZZ)V[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.$anonfun$mergeSchemasInParallel$2(ConvertToDeltaCommand.scala:542)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2(RDD.scala:863)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2$adapted(RDD.scala:863)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:131)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1491)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:750)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   [0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   Driver stacktrace:[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2450)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2399)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2398)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2398)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1156)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1156)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1156)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2638)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2580)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2569)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:938)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2224)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2245)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2264)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2289)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1030)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.withScope(RDD.scala:414)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.collect(RDD.scala:1029)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.mergeSchemasInParallel(ConvertToDeltaCommand.scala:528)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.getSchemaForBatch(ConvertToDeltaCommand.scala:575)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.$anonfun$inferSchema$3(ConvertToDeltaCommand.scala:592)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.$anonfun$inferSchema$3$adapted(ConvertToDeltaCommand.scala:586)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.collection.Iterator.foreach(Iterator.scala:943)[0m
[2024-07-18T18:46:34.573Z] [1m[31mE                   	at scala.collection.Iterator.foreach$(Iterator.scala:943)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.$anonfun$inferSchema$2(ConvertToDeltaCommand.scala:586)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:141)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:139)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.recordFrameProfile(ConvertToDeltaCommand.scala:449)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:134)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:77)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:67)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.recordOperation(ConvertToDeltaCommand.scala:449)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:133)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationForTablePath(DeltaLogging.scala:105)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationForTablePath$(DeltaLogging.scala:100)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.recordDeltaOperationForTablePath(ConvertToDeltaCommand.scala:449)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.inferSchema(ConvertToDeltaCommand.scala:586)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.tableSchema(ConvertToDeltaCommand.scala:473)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.$anonfun$performConvert$1(ConvertToDeltaCommand.scala:334)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:141)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:139)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.recordFrameProfile(ConvertToDeltaCommand.scala:71)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:134)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:77)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:67)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.recordOperation(ConvertToDeltaCommand.scala:71)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:133)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:123)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:111)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.recordDeltaOperation(ConvertToDeltaCommand.scala:71)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.performConvert(ConvertToDeltaCommand.scala:316)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ConvertToDeltaCommandBase.run(ConvertToDeltaCommand.scala:97)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:75)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:73)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.command.ExecutedCommandExec.executeCollect(commands.scala:84)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:97)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:97)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:93)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:481)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:82)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:481)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:30)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:267)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:263)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:30)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:457)[0m
[2024-07-18T18:46:34.574Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:93)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:80)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:78)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:219)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:96)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:618)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:613)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at sun.reflect.GeneratedMethodAccessor122.invoke(Unknown Source)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:282)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:750)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   Caused by: java.lang.NoSuchMethodError: org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ZZ)V[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.sql.delta.commands.ParquetTable.$anonfun$mergeSchemasInParallel$2(ConvertToDeltaCommand.scala:542)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2(RDD.scala:863)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2$adapted(RDD.scala:863)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:131)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1491)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2024-07-18T18:46:34.575Z] [1m[31mE                   	... 1 more[0m
```
</details>

--- a/integration_tests/src/main/python/delta_lake_test.py
+++ b/integration_tests/src/main/python/delta_lake_test.py
@@ -124,6 +124,8 @@ def test_delta_read_column_mapping(spark_tmp_path, reader_confs, mapping):
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake
 @ignore_order(local=True)
+@pytest.mark.skipif(not (is_databricks_runtime() or is_spark_340_or_later()), \
+                    reason="ParquetToSparkSchemaConverter changes not compatible with Delta Lake")
 def test_delta_name_column_mapping_no_field_ids(spark_tmp_path):
     data_path = spark_tmp_path + "/DELTA_DATA"
     def setup_parquet_table(spark):

